python/pyspark/sql/types.py

```diff
@@ -2214,12 +2211,9 @@ def verify_acceptable_types(obj: Any) -> None:
         # subclass of them can not be fromInternal in JVM
         if type(obj) not in _acceptable_types[_type]:
             raise PySparkTypeError(
-                error_class="CANNOT_ACCEPT_OBJECT_IN_TYPE",
-                message_parameters={
-                    "data_type": str(dataType),
-                    "obj_name": str(obj),
-                    "obj_type": type(obj).__name__,
-                },
```

Comment:
Shall we have an error class?

Reply:
Yes we can, but the error class would contain less meaningful contents, for example as below:

```python
  "CANNOT_ACCEPT_OBJECT_IN_TYPE": {
    "message": [
      "<message>"
    ]
  },
```

because we generate whole error message here.

Shall we add an error class anyway?

Reply:
Can we have the error class properly with fixing `new_msg` togehter? `new_msg` just appends `name` in front.  Also `object of ByteType out of range, got`, `object of ShortType out of range, got` has common messages too?

Reply:
Sure, will push another commit after addressing comments.

Reply:
CI finally turns to green